src/overlays/actors/ovl_En_Kz/z_en_kz.c

```diff
@@ -21,47 +28,458 @@ const ActorInit En_Kz_InitVars = {
+    Gameplay_ChangeCameraStatus(globalCtx, this->gameplayCamera, 7);
+    Gameplay_ClearCamera(globalCtx, this->cutsceneCamera);
+    func_8002DF54(globalCtx, &this->actor, 7);
+    this->actionFunc = EnKz_Wait;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kz/EnKz_Destroy.s")
+void EnKz_Wait(EnKz* this, GlobalContext* globalCtx) {
+    if (this->unk_1E0.unk_00 == 2) {
+        this->actionFunc = EnKz_SetupGetItem;
+        EnKz_SetupGetItem(this, globalCtx);
+    } else {
+        func_80034F54(globalCtx, &this->unk_2A6, &this->unk_2BE, 12);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kz/func_80A9D0C0.s")
+void EnKz_SetupGetItem(EnKz* this, GlobalContext* globalCtx) {
+    s32 getItemID;
+    f32 xzRange;
+    f32 yRange;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kz/func_80A9D130.s")
+    if (func_8002F410(this, globalCtx)) {
+        this->actor.attachedA = NULL;
+        this->unk_1E0.unk_00 = 1;
+        this->actionFunc = EnKz_StartTimer;
+    } else {
+        getItemID = this->isTrading == true ? GI_FROG : GI_TUNIC_ZORA;
+        yRange = fabsf(this->actor.yDistanceFromLink) + 1.0f;
+        xzRange = this->actor.xzDistanceFromLink + 1.0f;
+        func_8002F434(&this->actor, globalCtx, getItemID, xzRange, yRange);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kz/func_80A9D25C.s")
+void EnKz_StartTimer(EnKz* this, GlobalContext* globalCtx) {
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 6) && (func_80106BC8(globalCtx))) {
+        if (INV_CONTENT(ITEM_TRADE_ADULT) == ITEM_FROG) {
+            func_80088AA0(180); // start timer2 with 3 minutes
+            gSaveContext.eventInf[1] &= ~1;
+        }
+        this->unk_1E0.unk_00 = 0;
+        this->actionFunc = EnKz_Wait;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kz/func_80A9D3C8.s")
+void EnKz_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnKz* this = THIS;
+    s32 pad2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kz/func_80A9D42C.s")
+    if (LINK_IS_ADULT && !(gSaveContext.infTable[19] & 0x100)) {
+        gSaveContext.infTable[19] |= 0x100;
+    }
+    Collider_CylinderUpdate(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider);
+    SkelAnime_FrameUpdateMatrix(&this->skelanime);
+    EnKz_UpdateEyes(this);
+    Actor_MoveForward(&this->actor);
+    if (this->actionFunc != EnKz_StartTimer) {
+        func_80A9CB18(this, globalCtx);
+    }
+    this->actionFunc(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kz/func_80A9D490.s")
+s32 EnKz_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    s32 limb = limbIndex;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kz/func_80A9D520.s")
+    if (limb == 8 || limb == 9 || limb == 10) {
```

Comment:
same thing about parens around conditions

Reply:
so far we only use them when it makes sense to for readability. in this case it looks just fine without imo, and id say that adding them might make it even harder to read. i agree with the other two though

Reply:
Oh, really? I thought it was an "always" sort of thing. I'm much happier with it being not that way though.